### PR TITLE
Group color names together with alternative spellings, such as spaces/no-spaces and gray/grey

### PR DESCRIPTION
**Pull request by AlexKnauth**
Instead of writing a bunch of colors like this:
```racket
  ...
  ("darkblue" . #(0 0 139))
  ("darkcyan" . #(0 139 139))
  ("darkgoldenrod" . #(184 134 11))
  ("darkgray" . #(169 169 169))
  ("darkgrey" . #(169 169 169))
  ...
  ("dark blue" . #(0 0 139))
  ("dark cyan" . #(0 139 139))
  ("dark goldenrod" . #(184 134 11))
  ("dark gray" . #(169 169 169))
  ("dark grey" . #(169 169 169))
  ...
```

They can now be written like this:
```racket
  ...
  [(seq "dark" msp "blue")  #(0 0 139)]
  [(seq "dark" msp "cyan")  #(0 139 139)]
  [(seq "dark" msp "goldenrod")  #(184 134 11)]
  [(seq "dark" msp gray)  #(169 169 169)]
  ...
```
Where `msp` stands for "maybe space" and is defined as `(or " " "")`. Similarly `gray` is used for the alternative spellings of gray, and is defined as `(or "gray" "grey")`.

### CODE REVIEW

**Comment by rfindler** [comment]
I don't think we can change the set of names allowed here. But I was
planning to change the set for 2htdp/image.

Robby

**Comment by AlexKnauth** [comment]
Is there a reason why not?

**Comment by rfindler** [comment]
I think they are from various standards but I do not recall exactly. I
tried to search on the web but didn't get a lot of clarity.

Robby

**Comment by AlexKnauth** [comment]
Are the variations in spaces vs. no spaces from those standards?

**Comment by rfindler** [comment]
I think so but I am not sure. I think the the X11 spec is where they came
from and maybe it is in CSS too.

Robby

**Comment by AlexKnauth** [comment]
The HTML/CSS color names include some alternative spellings, like both `"gray"` and `"grey"`:
https://www.w3schools.com/colors/colors_names.asp

**Comment by rfindler** [comment]
This [wikipedia page](https://en.wikipedia.org/wiki/X11_color_names) seems to suggest that the color names in the X11 list (which is, I believe, where we got them) made their way into the CSS and SVG specifications. There are spaces in that list, but no with-space/without-space duplicates that I see.

Here is the [w3c color spec](https://www.w3.org/TR/css-color-3/), see the "4.3. Extended color keywords" section. It adds the gray/grey compared to the other (according to the text there; I didn't check the lists myself).

It looks like SVG 1.2 is the latest spec and its [color section](https://www.w3.org/TR/SVGColor12/) points to SVG 1.1 spec's [color keywords](https://www.w3.org/TR/SVG11/types.html#ColorKeywords). If I am reading SVG the SVG spec properly, no spaces are allowed in the color names.

And to make things yet more complicated, there's also a [long list of colors](https://en.wikipedia.org/wiki/Lists_of_colors) here.

In short, it seems like adding colors so that grey and gray all work has some precedent so perhaps we should consider it. Also it seems like the choices about where the spaces go is somewhat arbitrary, so maybe we should list them in the docs with spaces in "reasonable" places, but say that spaces are removed from the names before looking them up in the database (and then do that).

Not that I'm entirely sure about those two specific recommendations, but I'm certainly much less sure that we should try to add the list of colors in the long list of colors. 

Probably it would also be good to actually go over those four lists (ours, X11, SVG, CSS) and figure out what the actual differences are, if any.

**Comment by AlexKnauth** [comment]
Okay. It does make more sense to remove spaces from names before looking them up in the hash table, rather than have multiple copies of an entry in the hash table. The `find-color` method already calls `string-downcase` on the name before looking it up. It makes sense to normalize by remove spaces in the same place.

**Comment by mflatt** [comment]
FWIW, some internal history to add to Robby's thorough external history: The list in our implementation originally came from wxWindows. At that time, the numbers varied across platforms, and essentially the same list turned into a web de facto standard, so I tried to sync everything up (especially the numbers) and declare the list wouldn't change. I left the extra names with spaces in as a compromise to avoid breaking old programs. It looks like "grey" names were added to standards after I synced the colors, but I'm unsure.

I worry about changing anything here and my inclination would be to leave well enough alone. But it does sound ok to remove spaces on lookup strings and add "grey".

**Comment by rfindler** [comment]
As far as I know, we have no compelling reason to change these colors. Already, the `2htdp/image` color names are different and we can add the alternative grays there, which would have solved the original symptom behind this pull request. (I'm skeptical of changes that don't have a clear rationale, preferring to leave things alone if possible/reasonable.)

That said, pushing the changes that adjusts the implementation to be clearly in the cool way Alex suggests seems like a very nice win.

**Comment by AlexKnauth** [comment]
I have made https://github.com/racket/draw/pull/17, which just normalizes the spaces. I'm thinking of rebasing and modifying this pull request to only deal with adding `{gray, grey}` to every version of gray, since the spaces would be covered by https://github.com/racket/draw/pull/17, but I won't get to that until this evening.

**Comment by rfindler** [comment]
@AlexKnauth if I understand the consensus correctly here, it is to make no observable changes to the color names (but that cleaning up the code in the way you wrote looks quite cute). Are you saying you wish us to reach a different consensus?

**Comment by rfindler** [comment]
This doesn't seem to preserve existing behavior to me. It accepts things
with multiple spaces and spaces in strange places.

Robby

**Comment by rfindler** [comment]
Reading over the discussion here it seems like there was some confusion on the question of whether or not it was acceptable to have something like: `(send the-color-database find-color "b l u e")` change from returning `#f` to returning the same thing as `(send the-color-database find-color "blue")`. I understood that this was NOT acceptable but I may have been wrong about that. @AlexKnauth  seemed to believe that that was okay and desirable. @bennn and @mflatt what do you think?

**Comment by rfindler** [comment]
Looks like we should also be aware that #17 is in play.

**Comment by bennn** [comment]
Normalizing "b l u e" to "blue" is ok with me, as long as old colors have the same behavior. I think #17 did that, and this PR is only about how the source code looks.

I haven't read this PR carefully, but I can if that'd be helpful.
